### PR TITLE
Cosmetic fixes (wording, formatting)

### PR DESCRIPTION
**Pull request by lupino3**
This changes improves the wording around the ETW interface (formerly `nightmare.hpp`), and improves the formatting for the compilation README.

I also added vim files to .gitignore mostly to make my own git output clean.

I verified locally all the unit tests with VS2017 and they all work except for when_int32_values_are_same_is_should_match, which fails also in the case of a clean HEAD checkout, so I don't think this pull request is responsible in any way for that failure. I tried briefly to debug it but failed, might try to do it when I get some free time.

### CODE REVIEW

**Comment by lupino3** [comment]
@swannman please review this PR, I created it after our discussion yesterday. Thanks!

File: .gitignore [hunk 1]
```diff
@@ -38,3 +38,8 @@ githooks/*.dll
 
 # VS poops
```
Comment (swannman) [inline .gitignore]: could you remove "VS poops" while you're in here?

File: krabs/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Krabs Readme
+
+Important Preprocessor Definitions:
+
+* `UNICODE` - krabsetw expects the `UNICODE` preprocessor definition to be
+  defined. The code will not successfully compile without this flag. There is
+  no plan to support compilation without `UNICODE` being set.
+
+* `NDEBUG` - Set this varible in release builds to disable runtime type
```
Comment (swannman) [inline krabs/README.md]: typo `variable`

File: krabs/krabs.sln [hunk 3]
```diff
@@ -10,7 +10,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = ".nuget", ".nuget", "{5C9E22
 		.nuget\NuGet.targets = .nuget\NuGet.targets
 	EndProjectSection
 EndProject
-Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "O365.Security.Native.ETW", "..\O365.Security.Native.ETW\O365.Security.Native.ETW.vcxproj", "{ED4E6027-541F-440A-A5EE-15DBB7B89423}"
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "Microsoft.O365.Security.Native.ETW", "..\O365.Security.Native.ETW\O365.Security.Native.ETW.vcxproj", "{ED4E6027-541F-440A-A5EE-15DBB7B89423}"
```
Comment (swannman) [inline krabs/krabs.sln]: let's rename the `.vcxproj` to match the project name

File: .gitignore (hunk 1 above)
Comment (lupino3) [reply .gitignore]: Done, thanks for catching that.

File: krabs/README.md (hunk 2 above)
Comment (lupino3) [reply krabs/README.md]: Fixed.

File: krabs/krabs.sln (hunk 3 above)
Comment (lupino3) [reply krabs/krabs.sln]: I have done a more thorough rename, hope it's ok now.

**Comment by lupino3** [comment]
Thanks Matt! Could you please merge the PR?

**Pull request merged**